data/World/Ice Cavern.json

```diff
@@ -22,7 +25,22 @@
                 Blue_Fire and 
                 (is_adult or Slingshot or Sticks or 
                     Kokiri_Sword or can_use(Dins_Fire))",
+            "Ice Cavern Map Room Recovery Heart 1": "is_adult",
+            "Ice Cavern Map Room Recovery Heart 2": "is_adult",
+            "Ice Cavern Map Room Recovery Heart 3": "is_adult",
+            "Ice Cavern Block Room Red Rupee 1": "Blue_Fire and (can_play(Song_of_Time) or (is_child and can_use(Boomerang)))",
+            "Ice Cavern Block Room Red Rupee 2": "Blue_Fire and (can_play(Song_of_Time) or (is_child and can_use(Boomerang)))",
+            "Ice Cavern Block Room Red Rupee 3": "Blue_Fire and (can_play(Song_of_Time) or (is_child and can_use(Boomerang)))",
             "Ice Cavern Freestanding PoH": "Blue_Fire",
+            "Ice Cavern Hall Pot 1": "True",
+            "Ice Cavern Hall Pot 2": "True",
+            "Ice Cavern Spinning Blade Pot 1": "True",
+            "Ice Cavern Spinning Blade Pot 2": "True",
+            "Ice Cavern Spinning Blade Pot 3": "True",
+            "Ice Cavern Spinning Blade Flying Pot":  "True",
+            "Ice Cavern Near End Pot 1" : "Blue_Fire",
+            "Ice Cavern Near End Pot 2" : "Blue_Fire",
+            "Ice Cavern Frozen Pot" : "is_adult and Blue_Fire",
```

Comment:
It is so absurdly easy to get this item out of this pot while the ice still there that I'm not convinced it shouldn't just be in logic to do so. Sword alone can be a little iffy but just about everything else... Magic, bow, explosives, shield...